### PR TITLE
[macOS] Update template icons to Big Sur style

### PR DESCRIPTION
**Pull request by cbracken**
Updates the AppIcon assets in the macOS app template to meet Apple's
icon guidelines as of macOS Big Sur. Assets were moved to the
[flutter_template_images][1] package, which includes them as of version
4.1.0.

Ref: https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/app-icon/

Issue: https://github.com/flutter/flutter/issues/103371

[1]: https://github.com/flutter/packages/tree/main/packages/flutter_template_images

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by stuartmorgan-g — APPROVED** [review]
Looking at other icons in my dock that have the "logo on a blank blob" style (Chrome, VS Code, Discord), the padding here feels much higher. I would make the logo bigger to look more like those.

But it's also a placeholder, so I don't care that much. LGTM either way :)

**Review by stuartmorgan-g — CHANGES_REQUESTED** [review]
Wait, sorry, this shouldn't be in flutter/flutter. macOS must have gone in before the rules change.

You should put these in https://github.com/flutter/packages/tree/main/packages/flutter_template_images and update the macOS template to pull from there, like Windows does.

**Comment by cbracken** [comment]
Yep -- ended up asking @christopherfujino because I swore we'd created a better solution but couldn't find any reference to it on the wiki. I sent out https://github.com/flutter/packages/pull/1930 and will update this patch to a pubspec change and icon removal once that lands.

**Comment by cbracken** [comment]
Re: icon size, this was my basis for comparison:
image

**Comment by stuartmorgan** [comment]
I guess there's just no real consistency. 🤷🏻

**Comment by cbracken** [comment]
@stuartmorgan https://github.com/flutter/packages/pull/1930 has landed and was published as flutter_template_images 4.1.0. I've updated this patch to:
* Roll to latest packages
* Truncate and rename the existing images to end in `.img.tmpl`
* Update the template manifest to point to the `asset_size.png.img.tmpl` assets

File: packages/flutter_tools/templates/template_manifest.json [hunk 1]
```diff
@@ -98,13 +98,13 @@
...
         "templates/app_shared/macos.tmpl/Runner.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist",
         "templates/app_shared/macos.tmpl/Runner/AppDelegate.swift",
-        "templates/app_shared/macos.tmpl/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_1024.png",
-        "templates/app_shared/macos.tmpl/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_128.png",
-        "templates/app_shared/macos.tmpl/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_16.png",
-        "templates/app_shared/macos.tmpl/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_256.png",
-        "templates/app_shared/macos.tmpl/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_32.png",
-        "templates/app_shared/macos.tmpl/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_512.png",
-        "templates/app_shared/macos.tmpl/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_64.png",
+        "templates/app_shared/macos.tmpl/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_1024.png.img.tmpl",
```
Comment (Hixie) [inline packages/flutter_tools/templates/template_manifest.json:101]: can you add a comment on line 3 explaining the .img.tmpl stuff? e.g. telling people where to find the assets

**Comment by Hixie** [comment]
test-exempt: rolling dependencies

File: packages/flutter_tools/templates/template_manifest.json (hunk 1 above)
Comment (cbracken) [reply packages/flutter_tools/templates/template_manifest.json:101]: Ah will do -- this hit me too. I found a comment in template.dart in the tool, which wasn't the first place I looked :)

Comment (cbracken) [reply packages/flutter_tools/templates/template_manifest.json:101]: Done.

**Review by stuartmorgan-g — APPROVED** [review]
LGTM

Do we know why `charcode` is moving around in some pubspecs?

**Comment by cbracken** [comment]
> Do we know why charcode is moving around in some pubspecs?

I ran a full `flutter upgrade packages`, since IIRC we're generally encouraged not to manually update single packages but rather rely on the tool. It looks like charcode was removed as a transitive dependency; got a presubmit error asking me to remove it from the allowlist, which I did as part of this change. Updated the PR/commit description to account for this.

**Comment by cbracken** [comment]
I wonder if there's an "achievement unlocked" badge you can get for being broken by a recent parameter deprecation in a package that you wrote a decade ago (package:coverage) causing breakage in code that you wrote somewhere else a half-decade ago (coverage in the flutter tool).

No good deed goes unpunished...

**Comment by cbracken** [comment]
Alright the roll (without the updated `flutter_template_images` pin) has been extracted to https://github.com/flutter/flutter/pull/103771 and landed. I've rebased this on top of that such that this is just the remaining icon roll.

**Review by stuartmorgan-g — APPROVED** [review]
StillLGTM

**Pull request merged**